### PR TITLE
Add sortable table

### PR DESCRIPTION
**Pull request by buczi**
image
image

Code is pretty clunky, but is accomodating our need for sorting.

### CODE REVIEW

File: frontend/src/table/TableBody.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                const tData = data[accessor] ? data[accessor].toString().replaceAll('"','') : '---';
+                return <td key={uuidv4()}>{tData}</td>;
+              })}
+            </tr>
+          );
+        })}
+      </tbody>
+    );
+  };
+
```
Comment (MortonPL) [inline frontend/src/table/TableBody.tsx:40]: Files should end with an empty new line.

**Review by MortonPL — CHANGES_REQUESTED** [review]
JSX could use some formatting to improve readability of deeply nested blocks, but otherwise looks mostly OK in general. Left some comments on smaller things.

**Comment by MortonPL** [comment]
Closes #78.

File: frontend/src/table/SortedTable.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export const SortedTable: FunctionComponent<SortedTableProps> = ({ data, columns }) => {
+    const useSortableTable = (data: any[], columns: Column[]) => {
+    const handleSorting = (sortField: any, sortOrder: SortOrder) => {
+      if (sortField) {
+        const sorted = [...tableData].sort((a, b) => {
+          if (a[sortField] === null) return 1;
+          if (b[sortField] === null) return -1;
+          if (a[sortField] === null && b[sortField] === null) return 0;
+          return (
+            a[sortField].toString().localeCompare(b[sortField].toString(), "en", {
```
Comment (Dove6) [inline frontend/src/table/SortedTable.tsx:20]: What about sorting dates?

File: frontend/package.json
[the remark this answers is not in the document]
Comment (buczi) [reply frontend/package.json:1]: There was no other field valid for beeing a key, so in order to remove error from console 'non-original-key' or something like that i added this.

File: frontend/src/index.css [hunk 3]
```diff
@@ -11,3 +11,22 @@ code {
...
+
+.table th.up {
+  background-image: url("./imgs/up_arrow.png");
+}
+.table th.down {
+  background-image: url("./imgs/down_arrow.png");
+}
+.table th.default {
+  background-image: url("./imgs/default.png");
+}
```
[the remark this answers is not in the document]
Comment (buczi) [reply frontend/src/index.css:24]: I got them from the example repo.

File: frontend/src/documents/Documents.tsx [hunk 4]
```diff
@@ -63,64 +72,25 @@ export const Documents: FunctionComponent<DocumentsProps> = ({ apiClient }) => {
...
-        <Button variant="outline-secondary" onClick={() => navigateToVersionList(documentVersion.documentId)}>
-          Check versions
-        </Button>
-      </td>
-    </tr>
-  ));
-
+  const data: any[] = []
+  distinctByDocumentId(docsVersions).forEach(({ documentVersion, documentName }: DocumentNamedVersion, index: number) => {
+    data.push({
```
[the remark this answers is not in the document]
Comment (buczi) [reply frontend/src/documents/Documents.tsx:77]: Yea, so about that. It didn't work when i used map, so thats why this is done in such a manner. If you got suggestion how to fix it i will appriciate that.

File: frontend/src/table/SortedTable.tsx (hunk 2 above)
Comment (buczi) [reply frontend/src/table/SortedTable.tsx:20]: For our case works as intended.

File: frontend/src/documents/Documents.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Dove6) [reply frontend/src/documents/Documents.tsx:77]: Maybe you skipped parentheses around returned object? Parser treats it as function body in such a case. Here's some working code:

```tsx
  const data = distinctByDocumentId(docsVersions).map(({ documentVersion, documentName }: DocumentNamedVersion, index: number) => ({
    index: index + 1,
    version: documentName,
    name: documentVersion.versionName,
    created: getFormattedDate(documentVersion.createdAt),
    option: (
      <Button variant="outline-secondary" onClick={() => navigateToVersionList(documentVersion.documentId)}>
        Check versions
      </Button>
    )
  }));
```

[the remark this answers is not in the document]
Comment (buczi) [reply frontend/src/documents/Documents.tsx:77]: Will try it.

File: frontend/src/index.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (buczi) [reply frontend/src/index.css:24]: No info about that in the project.

File: frontend/src/table/SortedTable.tsx (hunk 2 above)
Comment (buczi) [reply frontend/src/table/SortedTable.tsx:33]: Fixed, for our case.

File: frontend/package.json
[the remark this answers is not in the document]
Comment (Dove6) [reply frontend/package.json:1]: [keys_without_uuidv4.txt](https://github.com/MiniaczQ/apsi/files/11873307/keys_without_uuidv4.txt)

File: frontend/src/index.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dove6) [reply frontend/src/index.css:24]: No license? Then usage is prohibited, strictly speaking. But well, we'll take care of it later.

**Review by MortonPL — APPROVED** [review]

**Pull request merged**